regression-tests.recursor-dnssec/recursortests.py

```diff
@@ -674,6 +668,9 @@ def tearDownResponders(cls):
 
     @classmethod
     def killProcess(cls, p):
+        # Don't try to kill it if it's already dead
+        if p.poll() is not None:
```

Comment:
I wonder it there are valid cases where the recursor is already gone at this point? (Note that this PR does not change anything in that regard since I believe we are already ignoring that case in the existing code).

Reply:
Yes, this is something to keep an eye on. If the startup succeeded, all processes we're trying to kill should be alive at the end. Any process not running is an indication of a crash or other issue.

But what I do not know is what happens if startup of one of the processes fails: does the rest of the startup continue? Will the teardown code be called?